### ISSUE TITLE
Create a repository for config server and client

### ISSUE THREAD

**Issue by kyessenov** [istio/pilot#844]
The upcoming evolution of the config system requires new APIs to standardize between mixer and pilot. This means that `istioctl` will no longer be compatible with older config servers. I propose we move all the operator clients (`apiserver`, `cmd/istioctl`) out of this repo, so that we can decouple  development, and avoid maintenance burden to keep `istioctl` compatible with `pilot` and `mixer` during the API churn. Once the uber-config backend is ready, all three clients can plug into the new backend at once.

@geeknoid @rshriram @louiscryan

**Comment by kyessenov** [comment]
also @sebastienvas : this would mean that istioctl is not guaranteed to work

**Comment by ldemailly** [comment]
> istioctl will no longer be compatible with older config servers.

I don't think backward/forward compatibility was ever a thing for alpha (nor should be until after first beta). version checks and negotiation is one of the requirements later though but not now

IMO we should keep trunk working and that includes istioctl (even if the implementation is in flux or changing)

**Comment by andraxylia** [comment]
There is no rush to remove istioctl from pilot now. We can copy it to the new repo, and it leave it here as well until we transition.

**Comment by kyessenov** [comment]
Fine, I'll have to live it here. It's weird that mixer can break their config API but we're forced to keep it in sync.
